test/src/test/java/hudson/cli/DefaultCliListenerTest.java

```diff
@@ -0,0 +1,168 @@
+ * THE SOFTWARE.
+ */
+
+package hudson.cli;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.hasSize;
+
+import hudson.cli.listeners.DefaultCliListener;
+import java.io.IOException;
+import java.util.List;
+import java.util.logging.Level;
+import jenkins.model.Jenkins;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.LoggerRule;
+import org.jvnet.hudson.test.MockAuthorizationStrategy;
+import org.jvnet.hudson.test.TestExtension;
+
+public class DefaultCliListenerTest {
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    @Rule
+    public LoggerRule logging = new LoggerRule();
+
+    private static final String USER = "cli-user";
+
+    @Before
+    public void setUp() throws IOException {
+        j.jenkins.setSecurityRealm(j.createDummySecurityRealm());
+        j.jenkins.setAuthorizationStrategy(new MockAuthorizationStrategy()
+                .grant(Jenkins.ADMINISTER)
+                .everywhere()
+                .to(USER));
+        j.createFreeStyleProject("p");
+        logging.record(DefaultCliListener.class, Level.FINE).capture(2);
+    }
+
+    @Test
+    public void commandExecutionSuccessIsLogged() throws Exception {
+        CLICommandInvoker command = new CLICommandInvoker(j, new ListJobsCommand());
+        command.asUser(USER).invoke();
+
+        List<String> messages = logging.getMessages();
+        assertThat(messages, hasSize(2));
+        assertThat(
+                messages.get(0),
+                containsString("Invoking CLI command list-jobs, with 0 arguments, as user %s.".formatted(USER)));
+        assertThat(
+                messages.get(1),
+                containsString(
+                        "Executed CLI command list-jobs, with 0 arguments, as user %s, return code 0".formatted(USER)));
+    }
+
+    @Test
+    public void commandExecutionErrorIsLogged() throws Exception {
+        CLICommandInvoker command = new CLICommandInvoker(j, new ListJobsCommand());
+        command.asUser(USER).invokeWithArgs("view-not-found");
+
+        List<String> messages = logging.getMessages();
+        assertThat(messages, hasSize(2));
+        assertThat(
+                messages.get(0),
+                containsString("Invoking CLI command list-jobs, with 1 arguments, as user %s.".formatted(USER)));
+        assertThat(
+                messages.get(1),
+                containsString("Failed call to CLI command list-jobs, with 1 arguments, as user %s.".formatted(USER)));
+        assertThat(
+                logging.getRecords().get(0).getThrown().getMessage(),
+                containsString("No view or item group with the given name 'view-not-found' found"));
+    }
+
+    @Test
+    public void commandExecutionUnexpectedErrorIsLogged() throws Exception {
```

Comment:
This test has no corresponding `actionExecutionUnexpectedErrorIsLogged`, why?

Reply:
you got me. I was unable to create a test implementation for a `@CLIMethod`, possibly due to a limitation in the CLICommandInvoker utility. Additionally, I couldn't find any similar tests.
Do you have any suggestions on how to trigger an exception in an existing default `@CLIMethod`?